### DISCUSSION TITLE
Successful Packaging But Not Working In MacOS

### DISCUSSION THREAD

**Discussion by jayden-vu** [pyinstaller/pyinstaller#5952] [PyInstaller]
When I run the module in the terminal it outputs the .app file And when I run it crashes and forces the login user and password to log back into macOS

**Comment by rokm** [comment]
This tells us practically nothing...

What is your program supposed to do and what packages it uses? 
Can you provide a minimum example that exhibits this problem? 
What version of macOS are you using (and is it on M1 by any chance)? 
What version of python and PyInstaller are you using?

Also, if the program manages the crash the user session, there should probably be some related messages in the macOS `Console` app...

**Reply by jayden-vu** [reply]
Hi, rokm!
- My program use packages tkinder, os, zipfile, json, shutil, re
- My Problem handing files in .zip and using UI to execute it
- Version MacOS Mojave version 10.14.6 using core i3
- Python 2.7.10 and python3 3.9.5   --- Pyinstaller 4.2

**Reply by rokm** [reply]
Does a tkinter-based hello world example work for you when frozen as .app bundle?

**Reply by jayden-vu** [reply]
I tried use pyinstaller with program hello world and it exported file .app but don't run and don't show any error

**Reply by jayden-vu** [reply]
icon in my program blinks and disappears

**Reply by rokm** [reply]
Can you paste the code for the hello world program that you are using?

**Reply by jayden-vu** [reply]
Source Code: test.py

```python
from sys import argv
import tkinter as tk
from tkinter import filedialog
from tkinter.constants import END
from tkinter import messagebox
class Application(tk.Frame):
    def __init__(self, master):
        super().__init__(master)
        self.master = master
        master.title('Part Manager')
        # Width height
        master.geometry("1125x550")
        # Add icon
        # master.iconbitmap('path')
        
        # Create widgets/grid
        self.create_widgets()


    def create_widgets(self):
        # Origin
        self.origin_text = tk.StringVar()
        self.origin_label = tk.Label(
            self.master, text='Origin Path Folder: ', font=('bold', 14), padx=10)
        self.origin_label.grid(row=0, column=0)
        self.origin_entry = tk.Entry(self.master, textvariable=self.origin_text, width=90)
        self.origin_entry.grid(row=0, column=1)

        #Button
        self.origin_add_btn = tk.Button(
        self.master, text="Add Part", width=12, command=lambda: self.add_folder('origin'))
        self.origin_add_btn.grid(row=0, column=2, pady=10, padx=10)
root = tk.Tk()
app = Application(master=root)
app.mainloop()
```

Terminal: pyinstaller --clean -w test.py
Output: test.app
Issue: When I click test.app icon of .app flashing and disappearing
I thinks version MacOS Mojave don't compatible python and module pyinstaller

**Reply by rokm** [reply]
It works on my test system, both with python 3.9.5 from python.org (intel version) and with homebrew python 3.9.5. Although with homebrew, you need to also install `python-tk` (i.e., `brew install python-tk@3.9`), otherwise `tkinter` is unavailable (both in python itself and consequently in frozen applications).

So whatever the problem is, it likely lies in your environment...

**Reply by rokm** [reply]
You can try running the non-app-bundle version (i.e., `dist/test/test`) from terminal and see if that works (and if not, what the error message is).
